### ISSUE TITLE
Regent: potential bug in parallelizer or reduction operator

### ISSUE THREAD

**Issue by mariodirenzo** [StanfordLegion/legion#505]
Compiling the following script
```
import "regent"

local C = regentlib.c

local Np = 10
local Nx = 2
local Ny = 2
local Nz = 2

fspace Fields {
   xx : double,
   y : double
}
__demand(__cuda, __leaf, __parallel)
task InitMesh( Mesh : region(ispace(int3d),Fields) )
where
   writes(Mesh.{xx,y})
do
   __demand(__openmp)
   for c in Mesh do
      Mesh[c].xx = double(c.x)/(Np-1)
      Mesh[c].y = 1.0/(Np*Np)
   end
end

__demand(__cuda, __leaf)
task InitStats( Mesh : region(ispace(int1d),Fields) )
where
   writes(Mesh.{xx,y})
do
   __demand(__openmp)
   for c in Mesh do
      Mesh[c].xx = double(c)/(Np-1)
      Mesh[c].y = 0.0
   end
end

__demand(__cuda, __leaf)
task AddStats(Mesh : region(ispace(int3d),Fields),
              Stat : region(ispace(int1d),Fields))
where
   reads(Mesh.y),
   reduces+(Stat.y)
do
   __demand(__openmp)
   for c in Mesh do
      Stat[c.x].y += Mesh[c].y
   end
end

task main()
   var r = region(ispace(int3d,{Np,Np,Np}),Fields)
   var s = region(ispace(int1d,Np),Fields)

   var cs = ispace(int1d,Nx)
   var ps = partition(equal, s, cs)

   var cr = ispace(int3d,{Nx,Ny,Nz})
   var pr = partition(equal, r, cr)

   InitMesh(r)
   InitStats(s)

   -- This parallel does not work
   __demand(__parallel)
   for c in cr do
      AddStats(pr[c], ps[c.x])
   end

end

--regentlib.start(main)
regentlib.saveobj(main, "Example.o", "object")
```
produces the error:
```
loop optimization failed: argument 2 is not provably projectable or invariant
         AddStats(pr[c], ps[c.x])
```

If the '' __demand(__parallel)'' in the main task is commented, the script is successfully compiled and correct results are produced.

Am I doing something wrong with the reduction privileges?

**Comment by elliottslaughter** [comment]
The problem is the use of `c.x` in the index expression of `ps[c.x]`. It should be possible to allow this, just will take a little time since everyone is crunching on SC papers at the moment.

**Comment by mariodirenzo** [comment]
Ok, thanks for replying.
Knowing that I am not doing anything massively wrong is already something.
Let me know when you will be able to implement this feature.

**Comment by elliottslaughter** [comment]
SC deadline is April 10, if you don't see a reply by a couple days after that, please ping this issue again.

**Comment by elliottslaughter** [comment]
Should be fixed in 2f18640cf081305939e833178a1ad25a92e590a5.

**Comment by mariodirenzo** [comment]
Yes, It works. Thanks
